src/update.c

```diff
@@ -105,10 +105,8 @@ static int check_manifests_uniqueness(int clrver, int mixver)
 static int update_loop(struct list *updates, struct manifest *server_manifest)
 {
 	int ret;
-	struct step step;
 
-	step = progress_get_step();
-	progress_set_step(step.current, "download_fullfiles");
+	progress_next_step("download_fullfiles", PROGRESS_BAR);
```

Comment:
@otaviobp  When the statedir is populated (previously existing files, pack extraction) this step reports 0 progress for ~20 seconds followed by an immediate jump to 100. This time is spent validating the hashes of fullfiles in the statedir and cache hits (no download) do not impact the progress bar. I feel this is too long to sit without feedback.

To solve this problem, we could improve the `check_files_hash` step to validate files in the statedir in addition to the files it checks at the target install locations. This step could create a list of files that we need to download (not already installed and not present in statedir) which could be passed to `download_fullfiles`. In this case, all files passed to `download_fullfiles` would require downloads.

Reply:
Easier for now would be to create another step inside download_fullfiles to process that. Do you think it will be a good solution?

Reply:
It would work, but it seems weird to have back-to-back file hash verifying steps.

Reply:
But I can accept that

Reply:
This is working well for the installer after adding a step.